src/program/program.controller.ts

```diff
@@ -5,28 +5,44 @@ import { AuthGuard } from '~/auth/auth.guard';
 import { Local } from '~/decorator';
 import { ProgramService } from './program.service';
 
-@UseGuards(AuthGuard)
 @Controller({
   path: 'programs',
   version: '2',
 })
 export class ProgramController {
   constructor(private programService: ProgramService) {}
 
-  @Get()
-  async getProgramByMemberId(@Local('member') member: JwtMember, @Req() request: Request) {
-    const { memberId } = request.query;
-
-    return this.programService.getProgramByMemberId(member.appId, String(memberId || member.memberId));
-  }
-
+  @UseGuards(AuthGuard)
   @Get('/expired')
   async getExpiredProgramByMemberId(@Local('member') member: JwtMember, @Req() request: Request) {
     const { memberId } = request.query;
 
     return this.programService.getExpiredProgramByMemberId(member.appId, String(memberId || member.memberId));
   }
 
+  @UseGuards(AuthGuard)
+  @Get('/:programId')
+  async getProgramByMemberId(@Local('member') member: JwtMember, @Param('programId') programId: string) {
+    const { permissions } = member;
+
+    const extraAllowPermission = ['PROGRAM_NORMAL'].find((e) => permissions.includes(e));
+    return ['PROGRAM_ADMIN'].some((e) => permissions.includes(e))
+      ? this.programService.getProgramByProgramId(programId)
+      : this.programService.getProgramByMemberId(member.memberId, programId, extraAllowPermission);
+  }
+
+  @Get('/:programId/contents/:programContentId/trial')
+  async getProgramContentById(@Param('programContentId') programContentId: string) {
+    const programContent = await this.programService.getProgramContentById(programContentId);
```

Comment:
add try catch to catch error 
if error cause , should throw api exception in controller layer

Reply:
FYI: https://docs.nestjs.com/exception-filters#binding-filters
`useGlobalFilters()`